pkg/codegen/nodejs/gen.go

```diff
@@ -2135,6 +2135,13 @@ func genNPMPackageMetadata(pkg *schema.Package, info NodePackageInfo) string {
 		scriptVersion = pluginVersion
 	}
 
+	pluginName := info.PluginName
+	// Default to the pulumi package name if PluginName isn't set by the user. This is different to the npm
```

Comment:
Note we could be a bit smarter here and only set name if needed. A lot of "@pulumi" packages would work even without name set here because we strip it off the npm name. That is a plugin "azure" with a npm package name of "@pulumi/azure" doesn't need the plugin name explicitly set. Similarly if the plugin name == the npm package name then this doesn't need to be set.

That would reduce the diff of this change a bit because all the test packages would fit this style, and also means things like actual azure codegen won't change from this.

But then maybe it's better to just be simple and explicit.

Reply:
I'm happy to take the bigger diff if we get simpler logic.